manifests/adopt.pp

```diff
@@ -47,22 +48,22 @@
 #   The name for the optional symlink in the installation directory.
 #
 define java::adopt (
-  $ensure         = 'present',
-  $version        = '8',
-  $version_major  = undef,
-  $version_minor  = undef,
-  $java           = 'jdk',
-  $proxy_server   = undef,
-  $proxy_type     = undef,
-  $url            = undef,
-  $basedir        = undef,
-  $manage_basedir = true,
-  $package_type   = undef,
-  $manage_symlink = false,
-  $symlink_name   = undef,
+  String[1]         $ensure         = 'present',
```

Comment:
If you check the $ensure case at the bottom of each file, the logic created for this parameter only handles two situations: either $ensure is set to 'present' and the code runs as intended; or $ensure is set to anything else and an error message is thrown stating that 'anything else' is not supported. Therefore, simply having a non-empty string assignation here will be enough to cover all available cases.

In an ideal scenario, the module would handle 'present' to install a package, 'absent' to uninstall it and '<anything else>' as a non-supported action but it looks like the uninstall feature was never fully implemented.

The datatype can be changed to an Enum['present', 'absent'] but then any value that is not it will throw a run-time error rather than being handled and showing a simple notice informing the user of a non-valid action.

Reply:
A value such as `Enum['present',Undef]` may be more appropriate then, as it either excepts present or nothing

Reply:
Undef values not accepted

Reply:
`Optional[Enum['present']]` will probably work

Reply:
Adding the Optional would require us to remove the default 'present' value for the parameter. Which in turn would make the parameter mandatory, which is kinda ironic tbh.

Enum['present'] by itself should do the trick tho